pkg/controllers/networkpolicy/cloudresource.go

```diff
@@ -31,148 +30,108 @@ const (
 )
 
 var (
-	resourceNPStatusSetter = map[securitygroup.CloudResourceType]func(tracker *cloudResourceNPTracker,
+	resourceNPStatusSetter = map[securitygroup.CloudResourceType]func(tracker *CloudResourceNpTracker,
 		reconciler *NetworkPolicyReconciler) (bool, error){
-		securitygroup.CloudResourceTypeVM: vmNPStatusSetter,
+		securitygroup.CloudResourceTypeVM: vmNpStatusSetter,
 	}
 )
 
 const (
 	AppliedSecurityGroupDeleteError = "Deleting/Detaching appliedTo sg %v: %v"
 )
 
-func vmNPStatusSetter(tracker *cloudResourceNPTracker, r *NetworkPolicyReconciler) (bool, error) {
-	log := r.Log.WithName("NPTracker")
-	status := tracker.computeNPStatus(r)
-	updated := false
-
-	vmItems, err := r.Inventory.GetVmFromIndexer(indexer.VirtualMachineByCloudId, tracker.cloudResource.Name)
-	if err != nil {
-		log.Error(err, "failed to get VM from VM cache")
-		return false, err
-	}
-	for _, item := range vmItems {
-		vm := item.(*runtimev1alpha1.VirtualMachine)
-		npStatus, ok := status[vm.Namespace]
-		if len(status[""]) > 0 {
-			if npStatus == nil {
-				npStatus = make(map[string]string)
-			}
-			for k, v := range status[""] {
-				npStatus[k] = v
-			}
+func vmNpStatusSetter(tracker *CloudResourceNpTracker, r *NetworkPolicyReconciler) (bool, error) { //nolint:unparam
```

Comment:
yes, otherwise it throws an error.
Other option is to modify the function signature. Since the function pointer is generic, did not want to update signature just for the Vmp.